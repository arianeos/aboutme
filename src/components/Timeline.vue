<template>
  <v-container class="pa-12" fluid id="timeline-container">
    <div class="text-center pa-12" v-show="fetching">
      <v-progress-circular
        :size="100"
        :width="7"
        color="purple"
        indeterminate
      ></v-progress-circular>
    </div>
    <v-timeline v-show="events.length > 0">
      <v-timeline-item
        v-for="event in events"
        :key="event.id"
        color="pink lighten-2"
        :icon="event.icon"
        fill-dot
        large
      >
        <v-card
          class="elevation-2"
          color="indigo lighten-2"
          style="color: white"
        >
          <v-card-title class="headline title">
            {{ event.title }}
          </v-card-title>
          <v-card-text class="white text--primary pa-6 card-text">
            {{ event.description }}
          </v-card-text>
        </v-card>
      </v-timeline-item>
    </v-timeline>
  </v-container>
</template>
<script>
import { mapState, mapActions } from "vuex";
export default {
  name: "Timeline",
  created() {
    this.loadEvents();
  },
  data() {
    return {
      fetching: false
    };
  },
  computed: {
    ...mapState("event", ["events"])
  },
  methods: {
    ...mapActions("event", ["fetchEvents"]),
    loadEvents() {
      this.fetching = true;
      this.fetchEvents()
        .then(() => (this.fetching = false))
        .catch(error => {
          this.fetching = false;
          console.error("An error occurred when fetching events", error);
        });
    }
  }
};
</script>
<style>
.card-text {
  font-size: 1rem;
}
</style>
