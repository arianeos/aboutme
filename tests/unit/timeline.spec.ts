import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex, { Store } from "vuex";
import Timeline from "@/components/Timeline.vue";
import Vuetify from "vuetify";
import Vue from "vue";

const localVue = createLocalVue();

Vue.use(Vuetify);
localVue.use(Vuex);

describe("Timeline", () => {
  let actions: any;
  let state;
  let store: Store<unknown>;

  beforeEach(() => {
    state = {
      events: [
        {
          id: 1,
          title: "Title 1",
          description: "Description 1",
          icon: "mdi-baby-carriage"
        },
        {
          id: 2,
          title: "Title 2",
          description: "Description 2",
          icon: "mdi-school"
        }
      ]
    };

    actions = {
      fetchEvents: jest.fn()
    };

    store = new Vuex.Store({
      modules: {
        event: {
          state,
          actions,
          namespaced: true
        }
      }
    });
  });

  it("should renders the component and matches snapshot", () => {
    const wrapper = shallowMount(Timeline, { store, localVue });
    expect(wrapper.html()).toMatchSnapshot();
    expect(actions.fetchEvents).toHaveBeenCalled();
  });
});
