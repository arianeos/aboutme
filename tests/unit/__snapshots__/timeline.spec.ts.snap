// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline should renders the component and matches snapshot 1`] = `
<v-container-stub id="timeline-container" tag="div" fluid="true" class="pa-12">
  <div class="text-center pa-12">
    <v-progress-circular-stub color="purple" indeterminate="true" rotate="0" size="100" width="7" value="0"></v-progress-circular-stub>
  </div>
  <v-timeline-stub>
    <v-timeline-item-stub color="pink lighten-2" filldot="true" icon="mdi-baby-carriage" large="true">
      <v-card-stub loaderheight="4" tag="div" color="indigo lighten-2" class="elevation-2" style="color: white;">
        <v-card-title-stub class="headline title">
          Title 1
        </v-card-title-stub>
        <v-card-text-stub class="white text--primary pa-6 card-text">
          Description 1
        </v-card-text-stub>
      </v-card-stub>
    </v-timeline-item-stub>
    <v-timeline-item-stub color="pink lighten-2" filldot="true" icon="mdi-school" large="true">
      <v-card-stub loaderheight="4" tag="div" color="indigo lighten-2" class="elevation-2" style="color: white;">
        <v-card-title-stub class="headline title">
          Title 2
        </v-card-title-stub>
        <v-card-text-stub class="white text--primary pa-6 card-text">
          Description 2
        </v-card-text-stub>
      </v-card-stub>
    </v-timeline-item-stub>
  </v-timeline-stub>
</v-container-stub>
`;
